packs/heritages/versatile-heritages/ganzi.json

```diff
@@ -18,11 +18,37 @@
                 "key": "FlatModifier",
                 "label": "PF2E.SkillVariant.VsControlled",
                 "predicate": [
-                    "controlled"
+                    "inflicts:controlled"
                 ],
                 "selector": "saving-throw",
                 "type": "circumstance",
                 "value": 1
+            },
+            {
+                "alwaysActive": true,
+                "key": "RollOption",
```

Comment:
Alternatively, could use an effect with a formula badge to randomly determine the resistance type.

Reply:
We have an effect as is, but there's a feat that lets them change the resistance. So some amount of control is needed.